src/multi.c

```diff
@@ -374,31 +374,42 @@ void touchWatchedKey(redisDb *db, robj *key) {
     }
 }
 
-/* On FLUSHDB or FLUSHALL all the watched keys that are present before the
- * flush but will be deleted as effect of the flushing operation should
- * be touched. "dbid" is the DB that's getting the flush. -1 if it is
- * a FLUSHALL operation (all the DBs flushed). */
-void touchWatchedKeysOnFlush(int dbid) {
-    listIter li1, li2;
+/* Set CLIENT_DIRTY_CAS to all clients of DB when DB is dirty.
+ * It may happen in the following situations:
+ * FLUSHDB, FLUSHALL, SWAPDB
+ *
+ * optionDb: for SWAPDB, the WATCH should be invalidated if
+ * the key exists in either of them, and skipped only if it
+ * doesn't exist in either. */
+void touchWatchedKeysOnDirty(long dbid, long *optionDb) {
```

Comment:
i know "doesn't exist in either" was my text. reading it again i think my English was broke 8-)
i think the following signature (function name and arguments will be easier to understand.
```suggestion
 * doesn't exist in both. */
void touchAllWatchedKeysInDb(redisDb * emptied, redisDb *replaced_with) {
```

if you take it, this change needs to be reflected in other places (including the argument name in the above comment)